.github/workflows/deploy.yml

```diff
@@ -0,0 +1,80 @@
+name: Build and Publish Storybook to GitHub Pages
+
+on:
+  push:
+    branches:
+      - develop
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+env:
+  STORYBOOK_RC_HOST: "https://demo.qa.rocket.chat"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "16.19.0"
+
+      - name: Install Dependencies
+        run: yarn
+
+      - name: Build Storybook
+        run: yarn build:storybook
+        working-directory: packages/react
+
+      - name: Build UI-Elements
+        run: yarn build:storybook
+        working-directory: packages/ui-elements
+
+      - name: Prepare Build Folder
+        run: |
+          mkdir -p build
+          mv -v packages/react/storybook-static/* build/
+          mkdir -p build/ui-elements
+          mv -v packages/ui-elements/storybook-static/* build/ui-elements/
+
+      - name: Upload Storybook Artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: storybooks
+          path: build/
+
+  deploy:
+    name: Deploy to GitHub Pages
+    runs-on: ubuntu-latest
+    needs:
+      - build
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: develop
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "16.19.0"
+
+      - uses: actions/download-artifact@v2
+        with:
+          name: storybooks
+          path: build
+
+      - name: Build and publish Storybook
+        uses: bitovi/github-actions-storybook-to-github-pages@v1.0.3
```

Comment:
If you are already building it manually with yarn build:storybook, it is unnecessary to build it again. Find a script that can avoid re-building and can be used only to publish the artifacts.

Check this out, how bitovi script works, might be helpful for you: https://github.com/bitovi/github-actions-storybook-to-github-pages/blob/main/action.yaml

Reply:
Okay, let me see that

Reply:
```
outputs:
  page_url:
    description: "The URL of the page"
    value: ${{ steps.deploy.outputs.page_url }}

- name: 'upload'
      uses: actions/upload-pages-artifact@v3
      with: 
        path: ${{ inputs.path }}

    - id: deploy
      name: Deploy to GitHub Pages
      uses: actions/deploy-pages@v4
      with:
        token: ${{ github.token }}
```

Try these with required modifications

Reply:
Worked!!!!

Reply:
checkout https://devanshkansagra.github.io/Embedded-Chat and https://devanshkansagra.github.io/Embedded-Chat/ui-elements

Reply:
and used only one job: build-and-deploy

Reply:
Awesome ! 

Please add your new commits to this PR then

Reply:
okay